vms/evm/subscriber.go

```diff
@@ -179,55 +117,55 @@ func (s *subscriber) ProcessFromHeight(height *big.Int, done chan bool) {
 	done <- true
 }
 
+func (s *subscriber) newWarpBlockInfo(header *types.Header) (*relayerTypes.WarpBlockInfo, error) {
+	var (
+		logs []types.Log
+		err  error
+	)
+	// Check if the block contains warp logs, and fetch them from the client if it does
+	if header.Bloom.Test(warpPrecompileLogFilter[:]) {
+		logs, err = s.ethClient.FilterLogs(context.Background(), interfaces.FilterQuery{
+			Topics:    [][]common.Hash{{warpPrecompileLogFilter}},
+			Addresses: []common.Address{warp.ContractAddress},
+			FromBlock: big.NewInt(int64(header.Number.Uint64())),
+			ToBlock:   big.NewInt(int64(header.Number.Uint64())),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &relayerTypes.WarpBlockInfo{
+		BlockNumber: header.Number.Uint64(),
+		WarpLogs:    logs,
+	}, nil
+}
+
 // Filter logs from the latest processed block to the latest block
 // Since initializationFilterQuery does not modify existing fields of warpFilterQuery,
 // we can safely reuse warpFilterQuery with only a shallow copy
 func (s *subscriber) processBlockRange(
-	ethClient ethclient.Client,
```

Comment:
Nit: In the function comment can we include that it process [fromBlock, toBlock] inclusive?